temporalio/worker/_workflow_instance.py

```diff
@@ -1791,6 +1800,42 @@ def _stack_trace(self) -> str:
             )
         return "\n\n".join(stacks)
 
+    def _enhanced_stack_trace(self) -> temporalio.common._EnhancedStackTrace:
+        sdk = temporalio.common._SDKInfo("sdk-python", __version__)
+
+        sources = dict()
+        stacks = []
+
+        for task in list(self._tasks):
+            locations = []
+            for frame in task.get_stack():
+                filename = frame.f_code.co_filename
+                line_number = frame.f_lineno
+                func_name = frame.f_code.co_name
+
+                try:
+                    source = inspect.getsourcelines(frame)
+                    code = "".join(source[0])
+                    line_number = int(source[1])
+                except OSError as ose:
+                    code = "Cannot access code.\n---\n%s" % ose.strerror
```

Comment:
Let's use interpolation instead of `%`, e.g.
```suggestion
                    code = f"Cannot access code.\n---\n{ose.strerror}"
```